core/src/main/java/org/elasticsearch/common/settings/KeyStoreWrapper.java

```diff
@@ -144,19 +172,33 @@ public static KeyStoreWrapper load(Path configDir) throws IOException {
         SimpleFSDirectory directory = new SimpleFSDirectory(configDir);
         try (IndexInput indexInput = directory.openInput(KEYSTORE_FILENAME, IOContext.READONCE)) {
             ChecksumIndexInput input = new BufferedChecksumIndexInput(indexInput);
-            CodecUtil.checkHeader(input, KEYSTORE_FILENAME, FORMAT_VERSION, FORMAT_VERSION);
+            int formatVersion = CodecUtil.checkHeader(input, KEYSTORE_FILENAME, MIN_FORMAT_VERSION, FORMAT_VERSION);
             byte hasPasswordByte = input.readByte();
             boolean hasPassword = hasPasswordByte == 1;
             if (hasPassword == false && hasPasswordByte != 0) {
                 throw new IllegalStateException("hasPassword boolean is corrupt: "
                     + String.format(Locale.ROOT, "%02x", hasPasswordByte));
             }
             String type = input.readString();
-            String secretKeyAlgo = input.readString();
+            String stringKeyAlgo = input.readString();
+            final String fileKeyAlgo;
+            if (formatVersion >= 2) {
```

Comment:
I'm not sure what that would help with. We have a constant for the min version supported, as well as the current version.  I could create a constant for "v1" and "v2", but how is reading a constant name different than seeing the version directly here? It would just mean `2` is in the name, so not really different?